pkg/querier/engine/query_tracker.go

```diff
@@ -17,35 +16,23 @@ import (
 
 // queryTracker implements promql.QueryTracker interface and can be used by PromQL engine for tracking queries.
 type queryTracker struct {
-	maxConcurrent int
-	sema          *semaphore.Weighted
-	tracker       *activitytracker.ActivityTracker
+	tracker *activitytracker.ActivityTracker
 }
 
 // newQueryTracker creates new queryTracker for use by PromQL engine. Activity tracker may be nil.
-func newQueryTracker(maxConcurrent int, tracker *activitytracker.ActivityTracker) *queryTracker {
-	semaIniValue := maxConcurrent
-	if semaIniValue <= 0 {
-		semaIniValue = math.MaxInt
-	}
+func newQueryTracker(tracker *activitytracker.ActivityTracker) *queryTracker {
 
 	return &queryTracker{
-		maxConcurrent: maxConcurrent,
-		sema:          semaphore.NewWeighted(int64(semaIniValue)),
-		tracker:       tracker,
+		tracker: tracker,
 	}
 }
 
 func (q *queryTracker) GetMaxConcurrent() int {
-	return q.maxConcurrent
+	// No limit.
+	return math.MaxInt
```

Comment:
This function is just used by Prometheus to set a metric value exposing the max concurrency. Prometheus is already using -1 when the activity tracker is not set at all (and so concurrency is unlimited), so I would use -1 here too.

```suggestion
	return -1
```